openmdao/core/tests/test_impl_comp.py

```diff
@@ -363,6 +365,35 @@ def test_list_residuals(self):
         self.assertEqual(text.count('value'), 0)
         self.assertEqual(text.count('resids'), 1)
 
+    def test_list_residuals_with_tol(self):
+        prob = om.Problem()
+        model = prob.model
+
+        model.add_subsystem('p1', om.IndepVarComp('x', 1.0))
+        model.add_subsystem('d1', SellarImplicitDis1())
+        model.add_subsystem('d2', SellarImplicitDis2())
+        model.connect('d1.y1', 'd2.y1')
+        model.connect('d2.y2', 'd1.y2')
+
+        model.nonlinear_solver = om.NewtonSolver(solve_subsystems=False)
+        model.nonlinear_solver.options['maxiter'] = 5
+        model.linear_solver = om.ScipyKrylov()
+        model.linear_solver.precon = om.LinearBlockGS()
+
+        prob.setup()
+        prob.set_solver_print(level=-1)
+
+        prob.run_model()
+
+        stdout = StringIO()
+        outputs = model.list_outputs(residuals_tol=0.01, residuals=True, out_stream=stdout)
+        text = stdout.getvalue().split('\n')
+        # P1 and D1 should not appear in the outputs section. This is being checked below
+        self.assertEqual(text[14], 'd2')
+        self.assertEqual(text[15], '  y2     [0.2323928]  [0.01679515]')
```

Comment:
This test fails for me with numerical differences:
```AssertionError: '  y2     [0.23238004]  [0.0167883]' != '  y2     [0.2323928]  [0.01679515]'
-   y2     [0.23238004]  [0.0167883]
?                  ---          ^^^
+   y2     [0.2323928]  [0.01679515]
?                 ++           ^^^^
```
I think you don't really care about the y values for the purposes of this test... 
I think rather you would just want to check that `d1` is present and `p1` and `d2` are not (due to their outputs being filtered)